Added tags and property_tags tables

euand1818 opened the pull request:
Added tags and property_tags tables. Both have public read access and authenticated write access. Write access may need thinking about, maybe only the user who created the property should be able to add tags?

euand1818 committed:
Added tags and property_tags tables, with public read access and auth write access

euand1818 committed:
updated types

OttoChu committed:
Merge branch 'development' into RRS-50-Update-Backend-to-store-tags

LukeHosk commented on supabase.types.ts:
```diff
@@ -290,6 +320,21 @@ export interface Database {
       }
+      tags: {
+        Row: {
+          tag_id: string
```
given that the `tag_name` field has unique & not null constraints, there is no real need for a separate id to index on

LukeHosk commented on supabase.types.ts:
```diff
@@ -189,6 +189,36 @@ export interface Database {
+            foreignKeyName: "property_tags_property_id_fkey"
+            columns: ["property_id"]
+            isOneToOne: false
+            referencedRelation: "properties"
```
i think the idea was supposed to be that tags are added to reviews, rather than directly to the properties. Then we can define a function to calculate how many of each tag has been set on reviews for a property.

LukeHosk committed:
update tag tables

- renamed `property_tags` to `review_tags` to reflect that tags are applied through a review
- removed the `tag_id` from the `tags` table as it can be indexed on the tags directly
- update the write policy for `review_tags` to only allow the user who created the review to edit tags linked to it

LukeHosk committed:
Merge branch 'development' into RRS-50-Update-Backend-to-store-tags
